sqlglot/expressions.py

```diff
@@ -1784,7 +1784,7 @@ def subquery(self, alias=None, copy=True) -> Subquery:
         instance = _maybe_copy(self, copy)
         return Subquery(
             this=instance,
-            alias=TableAlias(this=to_identifier(alias)),
+            alias=TableAlias(this=to_identifier(alias)) if alias else None,
```

Comment:
is this even valid to provide `None` as an identifier? if it is, should it be valid to provide an empty `TableAlias`?
The protection here seems a little too specific.

Reply:
> is this even valid to provide None as an identifier?

Yes, the corresponding expression argument is optional, so `None` means "don't include this when rendering sql". In the case of an empty `TableAlias`, it does render it but it produces an empty string, so the result is the same.

> should it be valid to provide an empty TableAlias

I'm not sure about this, cc @tobymao is there a reason both arguments are optional for `TableAlias`?

Reply:
it's not valid to provide an empty table alias